src/components/ColorPicker/components/ColorPickerContent/ColorPickerColorsGrid.jsx

```diff
@@ -22,11 +22,33 @@ export const ColorPickerColorsGrid = React.forwardRef(
       SelectedIndicatorIcon,
       colorSize,
       tooltipContentByColor,
-      colorShape
+      colorShape,
+      showCustomColorPicker,
+      initialCustomColor
     },
     ref
   ) => {
+    const [customColor, setCustomColor] = useState(initialCustomColor);
+    const [showColorPicker, setShowColorPicker] = useState();
+
     const getItemByIndex = useCallback(index => colorsToRender[index], [colorsToRender]);
+    const onCustomColorClicked = useCallback(
+      colorHex => {
+        onSelectionAction(colorsToRender.length);
+        setCustomColor(colorHex);
+        onColorClicked(colorHex, customColor);
+      },
+      [colorsToRender.length, customColor, onColorClicked, onSelectionAction]
+    );
+
+    const onClearSelectedColor = useCallback(
+      colorToRemove => {
+        setShowColorPicker();
+        setCustomColor();
+        onColorClicked(colorToRemove);
```

Comment:
won't this trigger a color selection for the custom color that we are currently removing?